File: BinaryTree/Leetcode101.java
public class Leetcode101 {
    public boolean isSymmetric(TreeNode root) {
        if(root == null)return true;
        return helper(root.left,root.right);
    }

    // we create a helper funtion that check inward and outward direction.
    public boolean helper(TreeNode left,TreeNode right){
        if(left == null && right == null)return true;
        if(left != null && right == null)return false;
        if(left == null && right != null)return false;
        if(left.val != right.val)return false;
        boolean leftRes = helper(left.left,right.right);
        boolean rightRes = helper(left.right,right.left);
        return leftRes && rightRes;
    }
}

File: BinarySearchTree/Python/Leetcode101.py
class Leetcode101:
    def isSymmetric(self, root: Optional[TreeNode]) -> bool:
        return self.checkSymmetric(root, root)

    def checkSymmetric(self, l: TreeNode, r: TreeNode) -> bool:
        if l is None and r is None:
            return True
        elif l is None or r is None:
            return False
        else: return l.val == r.val and self.checkSymmetric(l.left, r.right) and self.checkSymmetric(l.right, r.left)
